### PR TITLE
Add Spring for GraphQL version of tutorial

### PR DESCRIPTION
**Pull request by dondonz**
Now that Spring for GraphQL is GA, it's time to update the tutorial.

I've added the Spring for GraphQL version of the tutorial using features like controller annotations.

### CODE REVIEW

File: src/pages/tutorials/getting-started-with-spring-boot.md [hunk 1]
```diff
@@ -160,11 +152,179 @@ This schema defines one top level field (in the type `Query`):  `bookById` which
...
+      id
+      firstName
+      lastName
+    }
+  }
+}
+```
+
+You should see a response like this.
+![GraphQL response](/img/graphiQL.png)
```
Comment (bbakerman) [inline src/pages/tutorials/getting-started-with-spring-boot.md:301]: I think this image link is wrong - I tried to view it and it didnt show

File: src/pages/tutorials/getting-started-with-spring-boot.md [hunk 2]
```diff
@@ -160,11 +152,179 @@ This schema defines one top level field (in the type `Query`):  `bookById` which
...
-We create a new `GraphQLProvider` class in the package `com.graphqljava.tutorial.bookdetails` with an `init` method which will create a `GraphQL` instance:
+To help explain the key concepts, the following examples illustrate how to initialize GraphQL Java *without* Spring for GraphQL features.
+
+To make life a little easier, add [Google Guava](https://github.com/google/guava) to dependencies in `build.gradle`.
+```groovy
+dependencies {
+	implementation 'org.springframework.boot:spring-boot-starter-graphql'
+	implementation 'org.springframework.boot:spring-boot-starter-web'
+    implementation 'com.google.guava:guava:26.0-jre' // NEW
+	testImplementation 'org.springframework.boot:spring-boot-starter-test'
```
Comment (bbakerman) [inline src/pages/tutorials/getting-started-with-spring-boot.md:321]: did you want this indentation like this ... perhaps to call it out?

File: src/pages/tutorials/getting-started-with-spring-boot.md [hunk 3]
```diff
@@ -230,88 +391,15 @@ Overall the process of creating a `GraphQL` and `GraphQLSchema` instance looks l
 
 ![Creating GraphQL](/img/graphql_creation.png)
```
Comment (bbakerman) [inline src/pages/tutorials/getting-started-with-spring-boot.md:392]: do we have this image?

File: src/pages/tutorials/getting-started-with-spring-boot.md [hunk 4]
```diff
@@ -324,84 +412,40 @@ type Query {
...
-After starting it you will be asked for a URL, enter "http://localhost:8080/graphql".
-
-After that, you can query our example API and you should get back the result we mentioned above in the beginning. It should look something like this:
+A `PropertyDataFetcher` tries to lookup a property on a Java object in multiple ways. 
+In case of a `POJO` it assumes that for a GraphQL field `fieldX` it can find a POJO property called `fieldX`.
 
-![GraphQL Playground](/img/playground.png)
+## Further reading
+### Sample source code
+TODO upload this tutorial's sample code to the GraphQL Java org
```
Comment (bbakerman) [inline src/pages/tutorials/getting-started-with-spring-boot.md:440]: TODO

**Review by bbakerman — COMMENTED** [review]
Excellent

File: src/pages/tutorials/getting-started-with-spring-boot.md (hunk 2 above)
Comment (dondonz) [reply src/pages/tutorials/getting-started-with-spring-boot.md:321]: No I don't want this, thanks for the pickup!

File: src/pages/tutorials/getting-started-with-spring-boot.md (hunk 1 above)
Comment (dondonz) [reply src/pages/tutorials/getting-started-with-spring-boot.md:301]: This is a Docusaurus thing - you have to launch the site via `yarn start` to see the images, while in dev mode

File: src/pages/tutorials/getting-started-with-spring-boot.md (hunk 2 above)
Comment (dondonz) [reply src/pages/tutorials/getting-started-with-spring-boot.md]: Tab vs spaces strikes again

File: src/pages/tutorials/getting-started-with-spring-boot.md (hunk 3 above)
Comment (dondonz) [reply src/pages/tutorials/getting-started-with-spring-boot.md:392]: Yes the image is at `static/img/graphql_creation.png`. I'm reusing the old diagram

File: src/pages/tutorials/getting-started-with-spring-boot.md [hunk 5]
```diff
@@ -324,84 +413,38 @@ type Query {
 ...
 ```
 
-## Author DataFetcher
+### Author DataFetcher
+```java
+public DataFetcher getAuthorDataFetcher() {
```
Comment (andimarek) [inline src/pages/tutorials/getting-started-with-spring-boot.md:418]: I think that is outdated, we use not annotated methods?

Comment (dondonz) [reply src/pages/tutorials/getting-started-with-spring-boot.md:418]: As discussed I'll trim down the tutorial to just enough to get started

**Pull request merged**